lib/internal/async_hooks.js

```diff
@@ -179,9 +179,7 @@ function fatalError(e) {
 
 function lookupPublicResource(resource) {
   if (typeof resource !== 'object' || resource === null) return resource;
-  // TODO(addaleax): Merge this with owner_symbol and use it across all
-  // AsyncWrap instances.
```

Comment:
I think the bigger part of the todo here is the “use it across all AsyncWrap instances” bit, so maybe it makes sense to leave a comment of some kind in place :)

Reply:
I'm not entirely sure what `object()` is returning. Isn't the `object` being passed here the same as the JS wrapper object?
https://github.com/nodejs/node/blob/c975dff3c0f0f1ecb1574f3b10dd1d135a7704db/src/async_wrap.cc#L766-L768
How should we access the JS wrapper object?

Reply:
@raisinTen When I say “JS wrapper object”, I’m referring to the public API object, e.g. `net.Socket` when the C++ object is a `TCPWrap`. `object()` would return the `TCPWrap` instance in that case, not `net.Socket`.

Reply:
@addaleax Is there any way we can access the JS wrapper object of `object()` from the C++ side?

Reply:
@RaisinTen Err … that is the purpose of this whole thing here, `owner_symbol` and `resource_symbol` are both supposed to provide that, which is why they should be unified :sweat_smile:

Reply:
> I think the bigger part of the todo here is the “use it across all AsyncWrap instances” bit, so maybe it makes sense to leave a comment of some kind in place :)

@addaleax I have updated the PR to use the owner_symbol correctly across all the AsyncWrap instances.